code/modules/fishing/fishing_gear.dm

```diff
@@ -427,6 +427,11 @@ TYPEINFO(/obj/item/fish_portal)
 			W.set_loc(src)
 			W.dropped(user)
 
+	New()
+		..()
+		AddComponent(/datum/component/transfer_input/quickloading, allowed)
+		AddComponent(/datum/component/transfer_output)
```

Comment:
This component won't actually do anything unless you use the `TRANSFER_OR_DROP` macro for all outputted items. So instead of doing `item.set_loc(get_turf(src))` you'd do `TRANSFER_OR_DROP(item, get_turf(src))`

Reply:
The macro sends a signal to the `transfer_output` component with the item to transfer, then the component handles putting it there. Currently (line 391) it's just directly setting the location of the items to the turf it's sitting on, so the component isn't doing anything.

For clarity, the `/datum/component/transfer_input/quickloading` works fine but the `/datum/component/transfer_output` doesn't.

Reply:
x